x/mint/test/mint_test.go

```diff
@@ -0,0 +1,118 @@
+package test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/celestiaorg/celestia-app/app"
+	"github.com/celestiaorg/celestia-app/test/util/testnode"
+	minttypes "github.com/celestiaorg/celestia-app/x/mint/types"
+	"github.com/cosmos/cosmos-sdk/codec"
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	sdktypes "github.com/cosmos/cosmos-sdk/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	"github.com/stretchr/testify/suite"
+)
+
+type IntegrationTestSuite struct {
+	suite.Suite
+
+	accounts []string
+	cctx     testnode.Context
+}
+
+func (s *IntegrationTestSuite) SetupSuite() {
+	if testing.Short() {
+		s.T().Skip("skipping mint integration test in short mode.")
+	}
+
+	s.T().Log("setting up mint integration test suite")
+	s.accounts, s.cctx = testnode.DefaultNetwork(s.T())
```

Comment:
locally, I switched this to
```suggestion
	// set the minimum time between blocks to be 10 days. this will cause the
	// block time to increase by 10 days despite that time not actually passing,
	// which allows this test to not take 1 year to complete
	cparams := testnode.DefaultParams()
	cparams.Block.TimeIotaMs = int64(time.Hour.Milliseconds() * 240)

	cctx, _, _ := testnode.NewNetwork(t, cparams, testnode.DefaultTendermintConfig(), testnode.DefaultAppConfig())
	s.cctx = cctx
```
and got rid of accounts. This way we can set the node to not prune anything in the app config (so we can query the state of the chain at a given height) and increase the minimum time between blocks (TimeIota)

hehehe what a hack
```
height 1 time 2023-05-19 02:04:19.091578814 +0000 UTC
height 2 time 2023-05-29 02:04:19.091578814 +0000 UTC
height 3 time 2023-06-08 02:04:19.091578814 +0000 UTC
height 4 time 2023-06-18 02:04:19.091578814 +0000 UTC
height 5 time 2023-06-28 02:04:19.091578814 +0000 UTC
height 6 time 2023-07-08 02:04:19.091578814 +0000 UTC
height 7 time 2023-07-18 02:04:19.091578814 +0000 UTC
height 8 time 2023-07-28 02:04:19.091578814 +0000 UTC
height 9 time 2023-08-07 02:04:19.091578814 +0000 UTC
height 10 time 2023-08-17 02:04:19.091578814 +0000 UTC
height 11 time 2023-08-27 02:04:19.091578814 +0000 UTC
height 12 time 2023-09-06 02:04:19.091578814 +0000 UTC
height 13 time 2023-09-16 02:04:19.091578814 +0000 UTC
height 14 time 2023-09-26 02:04:19.091578814 +0000 UTC
height 15 time 2023-10-06 02:04:19.091578814 +0000 UTC
height 16 time 2023-10-16 02:04:19.091578814 +0000 UTC
height 17 time 2023-10-26 02:04:19.091578814 +0000 UTC
height 18 time 2023-11-05 02:04:19.091578814 +0000 UTC
height 19 time 2023-11-15 02:04:19.091578814 +0000 UTC
height 20 time 2023-11-25 02:04:19.091578814 +0000 UTC
height 21 time 2023-12-05 02:04:19.091578814 +0000 UTC
height 22 time 2023-12-15 02:04:19.091578814 +0000 UTC
height 23 time 2023-12-25 02:04:19.091578814 +0000 UTC
height 24 time 2024-01-04 02:04:19.091578814 +0000 UTC
height 25 time 2024-01-14 02:04:19.091578814 +0000 UTC
height 26 time 2024-01-24 02:04:19.091578814 +0000 UTC
height 27 time 2024-02-03 02:04:19.091578814 +0000 UTC
height 28 time 2024-02-13 02:04:19.091578814 +0000 UTC
height 29 time 2024-02-23 02:04:19.091578814 +0000 UTC
height 30 time 2024-03-04 02:04:19.091578814 +0000 UTC
height 31 time 2024-03-14 02:04:19.091578814 +0000 UTC
```